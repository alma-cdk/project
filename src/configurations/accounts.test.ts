import {
  AccountType,
  emptyMockAccountProps,
  AccountStrategy,
} from "./accounts";

test("AccountType", () => {
  expect(AccountType).toMatchSnapshot();
});

test("emptyMockAccountProps", () => {
  expect(emptyMockAccountProps).toMatchSnapshot();
});

describe("AccountStrategy", () => {
  test("one", () => {
    expect(
      AccountStrategy.one({
        shared: {
          id: "123456789012",
          config: {
            key: "value",
          },
        },
      }),
    ).toMatchSnapshot();
  });

  test("two", () => {
    expect(
      AccountStrategy.two({
        dev: {
          id: "123456789012",
          config: {
            foo: "bar",
          },
        },
        prod: {
          id: "213456789012",
          config: {
            baz: "quux",
          },
        },
      }),
    ).toMatchSnapshot();
  });

  test("three", () => {
    expect(
      AccountStrategy.three({
        dev: {
          id: "111111111111",
          config: {
            cidr: "172.16.0.0/22",
          },
        },
        preprod: {
          id: "222222222222",
          config: {
            cidr: "172.16.4.0/22",
          },
        },
        prod: {
          id: "333333333333",
          config: {
            cidr: "172.16.8.0/22",
          },
        },
      }),
    ).toMatchSnapshot();
  });
});
