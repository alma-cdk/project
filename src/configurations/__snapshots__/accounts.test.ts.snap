// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountStrategy one 1`] = `
Object {
  "mock": Object {
    "environments": Array [
      "mock[0-9]",
    ],
    "id": "123456789012",
  },
  "shared": Object {
    "config": Object {
      "key": "value",
    },
    "environments": Array [
      "development",
      "feature/[/a-zA-z0-9-]+",
      "test",
      "qa[0-9]",
      "staging",
      "preproduction",
      "production",
    ],
    "id": "123456789012",
  },
}
`;

exports[`AccountStrategy three 1`] = `
Object {
  "dev": Object {
    "config": Object {
      "cidr": "172.16.0.0/22",
    },
    "environments": Array [
      "development",
      "feature/[/a-zA-z0-9-]+",
      "test",
      "qa[0-9]",
      "staging",
    ],
    "id": "111111111111",
  },
  "mock": Object {
    "environments": Array [
      "mock[0-9]",
    ],
    "id": "123456789012",
  },
  "preprod": Object {
    "config": Object {
      "cidr": "172.16.4.0/22",
    },
    "environments": Array [
      "qa[0-9]",
      "preproduction",
    ],
    "id": "222222222222",
  },
  "prod": Object {
    "config": Object {
      "cidr": "172.16.8.0/22",
    },
    "environments": Array [
      "production",
    ],
    "id": "333333333333",
  },
}
`;

exports[`AccountStrategy two 1`] = `
Object {
  "dev": Object {
    "config": Object {
      "foo": "bar",
    },
    "environments": Array [
      "development",
      "feature/[/a-zA-z0-9-]+",
      "test",
      "qa[0-9]",
      "staging",
    ],
    "id": "123456789012",
  },
  "mock": Object {
    "environments": Array [
      "mock[0-9]",
    ],
    "id": "123456789012",
  },
  "prod": Object {
    "config": Object {
      "baz": "quux",
    },
    "environments": Array [
      "preproduction",
      "production",
    ],
    "id": "213456789012",
  },
}
`;

exports[`AccountType 1`] = `
Object {
  "DEV": "dev",
  "MOCK": "mock",
  "PREPROD": "preprod",
  "PROD": "prod",
  "SHARED": "shared",
}
`;

exports[`emptyMockAccountProps 1`] = `
Object {
  "id": "123456789012",
}
`;
